### PR TITLE
DEPR: Index.contains, DatetimeIndex.offset

### PR DESCRIPTION
**Pull request by jbrockmendel**
DatetimeIndex.offset I thought was removed in #29801, but apparently slipped through the cracks.

### CODE REVIEW

File: doc/source/whatsnew/v1.0.0.rst [hunk 1]
```diff
@@ -546,6 +546,8 @@ or ``matplotlib.Axes.plot``. See :ref:`plotting.formatters` for more.
 - Removed support for nested renaming in :meth:`DataFrame.aggregate`, :meth:`Series.aggregate`, :meth:`DataFrameGroupBy.aggregate`, :meth:`SeriesGroupBy.aggregate`, :meth:`Rolling.aggregate` (:issue:`18529`)
 - Passing ``datetime64`` data to :class:`TimedeltaIndex` or ``timedelta64`` data to ``DatetimeIndex`` now raises ``TypeError`` (:issue:`23539`, :issue:`23937`)
 - A tuple passed to :meth:`DataFrame.groupby` is now exclusively treated as a single key (:issue:`18314`)
+- Removed the previously deprecated :meth:`Index.contains`, use ``key in index`` instead (:issue:`30103`)
+- Removed the previously deprecated :attr:`RangeIndex._start`, :attr:`RangeIndex._stop`, :atttr:`RangeIndex._step` (:issue:`26581`)
```
Comment (jreback) [inline doc/source/whatsnew/v1.0.0.rst]: what is the min version of pyarrow that is ok here?

**Review by jreback — CHANGES_REQUESTED** [review]

File: doc/source/whatsnew/v1.0.0.rst (hunk 1 above)
Comment (jbrockmendel) [reply doc/source/whatsnew/v1.0.0.rst]: looks like the problem is in fastparquet; it fails in the most recent version 0.3.2

**Comment by jbrockmendel** [comment]
reverted the RangeIndex bits since fastparquet still uses them

**Comment by jorisvandenbossche** [comment]
Also, the start/stop/step were only at deprecation warning, we should maybe first elevate that to future warning?

**Review by jorisvandenbossche — CHANGES_REQUESTED** [review]
Can you add DatetimeIndex.offset to the whatsnew as well?

**Review by jreback — APPROVED** [review]
lgtm, ex @jorisvandenbossche comment

**Comment by jbrockmendel** [comment]
The offset note made it into #29801, just the actual removal fell through the cracks

**Comment by jbrockmendel** [comment]
> Also, the start/stop/step were only at deprecation warning, we should maybe first elevate that to future warning?

Seems reasonable.  Should we have a section in #6581 for DeprecationWarnings to upgrade to FutureWarnings? (or a separate tracker, but id prefer to keep it unified)

**Comment by jreback** [comment]
> > Also, the start/stop/step were only at deprecation warning, we should maybe first elevate that to future warning?
> 
> Seems reasonable. Should we have a section in #6581 for DeprecationWarnings to upgrade to FutureWarnings? (or a separate tracker, but id prefer to keep it unified)

we usually create a specific issue on that milestone, e.g. 1.1 or whatever (and/or can put on a unified tracker)

**Comment by jreback** [comment]
pls rebase

**Comment by jbrockmendel** [comment]
rebased + green

**Comment by jreback** [comment]
thanks @jbrockmendel

**Pull request merged**